### PR TITLE
[AutoPR- Security] Patch ignition-flatcar for CVE-2026-33814 [MEDIUM]

### PR DESCRIPTION
**Pull request by azurelinux-security**
Auto Patch ignition-flatcar for CVE-2026-33814.

Autosec pipeline run -> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=1116945&view=results

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, ``*-static`` subpackages, etc.) have had their ``Release`` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (``./cgmanifest.json``, ``./toolkit/scripts/toolchain/cgmanifest.json``, ``.github/workflows/cgmanifest.json``)
- [x] LICENSE-MAP files are up-to-date (``./LICENSES-AND-NOTICES/SPECS/data/licenses.json``, ``./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md``, ``./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON``)
- [x] All source files have up-to-date hashes in the ``*.signatures.json`` files
- [x] ``sudo make go-tidy-all`` and ``sudo make go-test-coverage`` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Auto Patch ignition-flatcar for CVE-2026-33814 (MEDIUM).

###### Change Log  

- CVE-2026-33814

###### Does this affect the toolchain?  

**YES/NO**

###### Associated issues  

- N/A

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2026-33814

###### Test Methodology

- Pipeline build id: [Buddy Build URL](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=1116957&view=results)

### CODE REVIEW

**Comment by azurelinux-security** [comment]
## 🔒 CVE Patch Review: CVE-2026-33814

**PR #17179** — [AutoPR- Security] Patch ignition-flatcar for CVE-2026-33814 [MEDIUM]
**Package:** ignition-flatcar | **Branch:** 3.0-dev

---

### Spec File Validation

| Check | Status | Detail |
|-------|--------|--------|
| Release bump | ✅ | Release bumped 3 → 4 |
| Patch entry | ✅ | Patch entries added: ['CVE-2026-33814.patch'] (covers ['CVE-2026-33814']) |
| Patch application | ✅ | %autosetup found in full spec — patches applied automatically |
| Changelog | ✅ | Changelog entry looks good |
| Signatures | ✅ | No source tarball changes — signatures N/A |
| Manifests | ✅ | Not a toolchain PR — manifests N/A |

---

### Build Verification

- **Build status:** ✅ PASSED
- **Artifact downloaded:** ✅
- **CVE applied during build:** ✅

#### 🤖 AI Build Log Analysis
- **Risk:** **low**
- **Summary:** The ignition-flatcar 2.22.0-4.azl3 package built successfully with the CVE-2026-33814 patch applied during %prep, the Go binary compiled without errors, and RPMs were produced. No compilation or linker failures occurred; only minor, non-fatal packaging warnings were observed.
- **AI-detected warnings:**
  - rpmbuild warning: Could not canonicalize hostname: 9904668bc000000
  - find-debuginfo/debugedit message: /bin/debugedit: .../usr/bin/ignition: DWARF version 0 unhandled (typical for Go binaries; debuginfo extraction may be limited)

---

#### 🧪 Test Log Analysis

- **Test status:** ✅ PASSED

##### 🤖 AI Test Log Analysis
- **Risk:** **medium**
- **Summary:** The ignition-flatcar test suite ran successfully with no failures. The CVE-2026-33814 patch (along with two other CVE patches) applied cleanly, and all executed Go package tests reported "ok" where present. The suite was modified to skip non-security checks (gofmt/gofix), and it concluded with EXIT STATUS 0.

---

### Patch Analysis

- **Match type:** **backport**
- **Risk assessment:** **low**
- **Summary:** The PR applies the same functional change as upstream to vendor/golang.org/x/net/http2/transport.go: it validates every HTTP/2 SETTINGS parameter by calling s.Valid() at the start of ForeachSetting and removes the now-redundant per-setting validation in the SettingEnableConnectProtocol branch. The upstream test addition is not included, but the core fix is present and equivalent.
- **Missing hunks:**
  - Addition of TestTransportDoNotHangOnZeroMaxFrameSize in http2/transport_test.go to verify the fix prevents hanging on zero max frame size.

<details>
<summary>Detailed analysis</summary>

Comparison of the code changes shows the PR introduces the same logic as the upstream patch within clientConnReadLoop.processSettingsNoWrite: it adds an early call to s.Valid() with error propagation for all SETTINGS values and removes the redundant validity check inside the SettingEnableConnectProtocol case. Context line offsets differ slightly (expected for a vendored file), and the path reflects a vendor directory (vendor/golang.org/x/net/http2/transport.go) rather than the upstream module path, consistent with a distro backport. No other functional differences are present in the code hunk. The upstream patch also added a test (TestTransportDoNotHangOnZeroMaxFrameSize) to ensure a zero SettingMaxFrameSize no longer causes a hang; this test is not part of the PR, which is typical in packaging patches. Functionally, the PR should prevent hangs by rejecting invalid SETTINGS frames early, aligning behavior with upstream. The change is minimal and targets validation; regression risk is low, with the only behavioral change being earlier rejection of invalid SETTINGS values, which matches upstream intent and CVE fix.

</details>

<details>
<summary>Raw diff (upstream vs PR)</summary>

```diff
--- upstream
+++ pr
@@ -1,68 +1,54 @@
-From 1e71bd86e4a302b4e731bc06da6eb51679c7bd49 Mon Sep 17 00:00:00 2001
-From: "Nicholas S. Husin" <nsh@golang.org>
-Date: Tue, 31 Mar 2026 15:15:30 -0400
-Subject: [PATCH] http2: prevent hanging Transport due to bad SETTINGS frame
-
-This CL backports https://go.dev/cl/761581 to x/net.
-
-Fixes golang/go#78476
-Fixes CVE-2026-33814
-
-Change-Id: Ied435a51fdd8664d41dae14d082c39c76a6a6964
-Reviewed-on: https://go-review.googlesource.com/c/net/+/761640
-LUCI-TryBot-Result: Go LUCI <golang-scoped@luci-project-accounts.iam.gserviceaccount.com>
-Reviewed-by: Nicholas Husin <husin@google.com>
-Reviewed-by: Damien Neil <dneil@google.com>
----
- http2/transport.go      |  6 +++---
- http2/transport_test.go | 13 +++++++++++++
- 2 files changed, 16 insertions(+), 3 deletions(-)
-
-diff --git a/http2/transport.go b/http2/transport.go
-index 2e9c2f6a52..8132233310 100644
---- a/http2/transport.go
-+++ b/http2/transport.go
-@@ -2861,6 +2861,9 @@ func (rl *clientConnReadLoop) processSettingsNoWrite(f *SettingsFrame) error {
- 
- 	var seenMaxConcurrentStreams bool
- 	err := f.ForeachSetting(func(s Setting) error {
-+		if err := s.Valid(); err != nil {
-+			return err
-+		}
- 		switch s.ID {
- 		case SettingMaxFrameSize:
- 			cc.maxFrameSize = s.Val
-@@ -2892,9 +2895,6 @@ func (rl *clientConnReadLoop) processSettingsNoWrite(f *SettingsFrame) error {
- 			cc.henc.SetMaxDynamicTableSize(s.Val)
- 			cc.peerMaxHeaderTableSize = s.Val
- 		case SettingEnableConnectProtocol:
--			if err := s.Valid(); err != nil {
--				return err
--			}
- 			// If the peer wants to send us SETTINGS_ENABLE_CONNECT_PROTOCOL,
- 			// we require that it do so in the first SETTINGS frame.
- 			//
-diff --git a/http2/transport_test.go b/http2/transport_test.go
-index d948b881a3..9068375177 100644
---- a/http2/transport_test.go
-+++ b/http2/transport_test.go
-@@ -5677,6 +5677,19 @@ func testTransportTLSNextProtoConnImmediateFailureUnused(t testing.TB) {
- 	}
- }
- 
-+func TestTransportDoNotHangOnZeroMaxFrameSize(t *testing.T) {
-+	synctestTest(t, testTransportDoNotHangOnZeroMaxFrameSize)
-+}
-+func testTransportDoNotHangOnZeroMaxFrameSize(t testing.TB) {
-+	tc := newTestClientConn(t)
-+	tc.writeSettings(Setting{ID: SettingMaxFrameSize, Val: 0})
-+	tc.wantFrameType(FrameSettings)
+diff --git a/SPECS/ignition-flatcar/CVE-2026-33814.patch b/SPECS/ignition-flatcar/CVE-2026-33814.patch
+new file mode 100644
+index 00000000000..b2bb6316b23
+--- /dev/null
++++ b/SPECS/ignition-flatcar/CVE-2026-33814.patch
+@@ -0,0 +1,48 @@
++From 0b1ee6aa7b6db455fe929e37650ced71ec62ec10 Mon Sep 17 00:00:00 2001
++From: "Nicholas S. Husin" <nsh@golang.org>
++Date: Tue, 31 Mar 2026 15:15:30 -0400
++Subject: [PATCH] http2: prevent hanging Transport due to bad SETTINGS frame
 +
-+	req, _ := http.NewRequest("POST", "https://dummy.tld/", strings.NewReader("body"))
-+	tc.roundTrip(req)
-+	// Previously, https://go.dev/issue/78476 caused an infinite hang here.
-+}
++This CL backports https://go.dev/cl/761581 to x/net.
 +
- func TestExtendedConnectClientWithServerSupport(t *testing.T) {
- 	SetDisableExtendedConnectProtocol(t, false)
- 	ts := newTestServer(t, func(w http.ResponseWriter, r *http.Request) {
++Fixes golang/go#78476
++Fixes CVE-2026-33814
++
++Change-Id: Ied435a51fdd8664d41dae14d082c39c76a6a6964
++Reviewed-on: https://go-review.googlesource.com/c/net/+/761640
++LUCI-TryBot-Result: Go LUCI <golang-scoped@luci-project-accounts.iam.gserviceaccount.com>
++Reviewed-by: Nicholas Husin <husin@google.com>
++Reviewed-by: Damien Neil <dneil@google.com>
++Signed-off-by: Azure Linux Security Servicing Account <azurelinux-security@microsoft.com>
++Upstream-reference: https://github.com/golang/net/commit/1e71bd86e4a302b4e731bc06da6eb51679c7bd49.patch
++---
++ vendor/golang.org/x/net/http2/transport.go | 6 +++---
++ 1 file changed, 3 insertions(+), 3 deletions(-)
++
++diff --git a/vendor/golang.org/x/net/http2/transport.go b/vendor/golang.org/x/net/http2/transport.go
++index f26356b..a401b27 100644
++--- a/vendor/golang.org/x/net/http2/transport.go
+++++ b/vendor/golang.org/x/net/http2/transport.go
++@@ -2858,6 +2858,9 @@ func (rl *clientConnReadLoop) processSettingsNoWrite(f *SettingsFrame) error {
++ 
++ 	var seenMaxConcurrentStreams bool
++ 	err := f.ForeachSetting(func(s Setting) error {
+++		if err := s.Valid(); err != nil {
+++			return err
+++		}
++ 		switch s.ID {
++ 		case SettingMaxFrameSize:
++ 			cc.maxFrameSize = s.Val
++@@ -2889,9 +2892,6 @@ func (rl *clientConnReadLoop) processSettingsNoWrite(f *SettingsFrame) error {
++ 			cc.henc.SetMaxDynamicTableSize(s.Val)
++ 			cc.peerMaxHeaderTableSize = s.Val
++ 		case SettingEnableConnectProtocol:
++-			if err := s.Valid(); err != nil {
++-				return err
++-			}
++ 			// If the peer wants to send us SETTINGS_ENABLE_CONNECT_PROTOCOL,
++ 			// we require that it do so in the first SETTINGS frame.
++ 			//
++-- 
++2.45.4
++

```

</details>

---

### Verdict

✅ **APPROVED** — All checks passed. Ready to merge.

**Review by Kanishk-Bansal — APPROVED** [review]
Patch matches upstream, `transport_test.go` is omitted as test file is not present in src tarball.

- [x] Buddy Build 
- [x] patch applied during the build (check `rpm.log`)
- [x] patch include an upstream reference
- [x] PR has security tag

**Review by kgodara912 — APPROVED** [review]
Patch matches with upstream reference except test file. PR check is known failure. Buddy build is successful. LGTM.
